query/src/main/java/io/keen/client/java/MultiAnalysis.java

```diff
@@ -0,0 +1,172 @@
+        if (null == this.analyses || this.analyses.isEmpty()) {
+            throw new IllegalArgumentException("MultiAnalysis requires that 'analyses' be set.");
+        }
+    }
+
+    /**
+     * This builder class helps configure the required and optional parameters that are appropriate
+     * for a MultiAnalysis.
+     */
+    public static class Builder extends CollectionAnalysis.Builder<Builder> {
+        // required
+        private Collection<SubAnalysis> subAnalyses;
+
+
+        /**
+         * Get the collection of sub-analyses for this multi-analysis.
+         *
+         * @return The sub-analyses.
+         */
+        public Collection<SubAnalysis> getSubAnalyses() { return this.subAnalyses; }
+
+
+        // Configurable properties added by this builder.
+
+        /**
+         * Set the collection of sub-analyses. Existing sub-analyses will be discarded.
+         *
+         * @param subAnalyses The new analysis arguments to replace the discarded ones, if any.
+         */
+        public void setSubAnalyses(Collection<? extends SubAnalysis> subAnalyses) {
+            this.subAnalyses = null;
+
+            // Client code may just be clearing all the analyses.
+            if (null != subAnalyses) {
+                withSubAnalyses(subAnalyses);
+            }
+        }
+
+        /**
+         * Adds a collection of sub-analyses.
+         *
+         * @param subAnalyses The new analysis arguments to add to the existing analyses.
+         * @return This instance (for method chaining).
+         */
+        public Builder withSubAnalyses(Collection<? extends SubAnalysis> subAnalyses) {
+            if (null == subAnalyses) {
+                throw new IllegalArgumentException("The 'subAnalyses' parameter cannot be null.");
+            }
+
+            // Add each sub-analysis to the list, appending to anything that already exists.
+            for (SubAnalysis subAnalysis : subAnalyses) {
+                addSubAnalysis(subAnalysis);
+            }
+
+            return this;
+        }
+
+        /**
+         * Adds a sub-analysis to the existing analyses.
+         *
+         * @param subAnalysis The sub-analysis to add to the existing analyses.
+         * @return This instance (for method chaining).
+         */
+        public Builder withSubAnalysis(SubAnalysis subAnalysis) {
+            addSubAnalysis(subAnalysis);
+            return this;
+        }
+
+        /**
+         * Adds a group by property to the list of existing group by properties.
+         *
+         * @param subAnalysis TThe sub-analysis to add to the existing analyses.
+         */
+        public void addSubAnalysis(SubAnalysis subAnalysis) {
+            if (null == subAnalysis) {
+                throw new IllegalArgumentException("The 'subAnalysis' parameter cannot be null.");
+            }
+
+            if (null == this.subAnalyses) {
+                // No point in specifying one twice. Though obviously for now, since SubAnalysis
```

Comment:
Maybe update the comment to make that clear. Right now it reads a bit jarringly for me as well. Prepending it with something like "Using a HashSet here, because it will only allow a single subAnalyses with the same name"